### PR TITLE
feat(newsletter): add root state slice interface

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

#1595

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

selector types have changed

## Other information

### CODE REVIEW

**Comment by lderrickable** [comment]
What made you think to add this feature?

File: libs/newsletter/state/src/reducers/newsletter.reducer.ts [hunk 1]
```diff
@@ -5,13 +5,18 @@ import {
...
 } from './../actions/newsletter.actions';
+import { DAFF_NEWSLETTER_STORE_FEATURE_KEY } from './newsletter-store-feature-key';
 
 export interface DaffNewsletterState {
   success: boolean;
   loading: boolean;
   error: DaffStateError;
 }
 
+export interface DaffContactStateRootSlice {
```
Comment (damienwebdev) [inline libs/newsletter/state/src/reducers/newsletter.reducer.ts:16]: The slice name is a copy paste error.

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/newsletter/state/src/reducers/newsletter.reducer.ts (hunk 1 above)
Comment (griest024) [reply libs/newsletter/state/src/reducers/newsletter.reducer.ts]: fixed

**Comment by griest024** [comment]
> What made you think to add this feature?

I prefer to type things as strictly as possible. Adding these types to the selectors created a lot of type errors elsewhere in the codebase that wouldn't have otherwise shown up. `Record<string, any>` is a BS type, though it was at least better than `object`.

**Review by damienwebdev — APPROVED** [review]
lgtm

**Pull request merged**